superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent/index.tsx

```diff
@@ -141,12 +139,8 @@ export const useSimpleTabFilterProps = (props: Props) => {
       );
     }
     return !(
-      (props.datasource.type === 'druid' &&
-        TABLE_ONLY_OPERATORS.indexOf(operator) >= 0) ||
-      (props.datasource.type === 'table' &&
-        DRUID_ONLY_OPERATORS.indexOf(operator) >= 0) ||
-      (props.adhocFilter.clause === CLAUSES.HAVING &&
-        HAVING_OPERATORS.indexOf(operator) === -1)
+      props.adhocFilter.clause === CLAUSES.HAVING &&
+      HAVING_OPERATORS.indexOf(operator) === -1
```

Comment:
Might be cleaner to write as:

```js
return (
    props.adhocFilter.clause !== CLAUSES.HAVING ||
    HAVING_OPERATORS.indexOf(operator) !== -1
);
```

Reply:
I like that thank you!

Reply:
You can always replace `!(a && b)` with `!a || !b`, and in this case I think it makes it easier to follow the logic.